modules/programs/git-worktree-switcher.nix

```diff
@@ -0,0 +1,39 @@
+{ pkgs, config, lib, ... }:
+
+let
+  inherit (lib) mkEnableOption mkPackageOption optionalString;
+
+  cfg = config.programs.git-worktree-switcher;
+
+  initScript = shell:
+    if (shell == "fish") then ''
+      ${lib.getExe pkgs.git-worktree-switcher} init ${shell} | source
+    '' else ''
+      eval "$(${lib.getExe pkgs.git-worktree-switcher} init ${shell})"
+    '';
+in {
+  meta.maintainers = with lib.maintainers; [ jiriks74 mateusauler ];
+
+  options.programs.git-worktree-switcher = {
+    enable = mkEnableOption "git-worktree-switcher";
+    package = mkPackageOption pkgs "git-worktree-switcher" { };
+    enableBashIntegration =
+      lib.hm.shell.mkBashIntegrationOption { inherit config; };
+
+    enableFishIntegration =
+      lib.hm.shell.mkFishIntegrationOption { inherit config; };
+
+    enableZshIntegration =
+      lib.hm.shell.mkZshIntegrationOption { inherit config; };
+  };
+
+  config = lib.mkIf cfg.enable {
+    home.packages = [ cfg.package ];
+    programs.bash.initExtra =
+      optionalString cfg.enableBashIntegration (initScript "bash");
+    programs.fish.interactiveShellInit =
+      optionalString cfg.enableFishIntegration (initScript "fish");
+    programs.zsh.initExtra =
+      optionalString cfg.enableZshIntegration (initScript "zsh");
```

Comment:
This should not be the job of home manager, but instead packaged with `installShellCompletion`.

In fact it [already is](https://github.com/NixOS/nixpkgs/blob/64e75cd44acf21c7933d61d7721e812eac1b5a0a/pkgs/by-name/gi/git-worktree-switcher/package.nix#L47-L49). Making this option redundant.

Reply:
Opened #6440.

Reply:
As mentioned in https://github.com/nix-community/home-manager/pull/6440, the integrations are loading a function into the current shell which is necessary for this utility to work (to avoid creating subshells when changing directories) and therefore shouldn't be removed. This module works on the same ideas as others - allowing users to easily enable the utility in their Nix/Home-Manager configurations without creating a custom `.<insert_shell_here>rc` or `programs.<insert_shell_here>.initExtra`.

See
https://github.com/nix-community/home-manager/blob/83bd3a26ac0526ae04fa74df46738bb44b89dcdd/modules/programs/thefuck.nix#L37-L43